x-pack/test/functional/config.js

```diff
@@ -284,6 +284,20 @@ export default async function ({ readConfigFile }) {
           ],
         },
 
+        global_upgrade_assistant_role: {
+          elasticsearch: {
+            cluster: ['manage'],
```

Comment:
As discussed offline, I think this will fix the test (by resolving the 403 error you are seeing on `/api/upgrade_assistant/status`):

```suggestion
            cluster: ['manage'],
            indices: [
              {
                names: ['apm-*'],
                privileges: ['read', 'view_index_metadata'],
                field_security: { grant: ['*'], except: [] },
              },
            ],
```